src/integrationTest/java/com/serjltt/moshi/adapters/LazyAdaptersAutoValueTest.java

```diff
@@ -45,4 +47,21 @@
     assertThat(fromJson.meta().value1).isEqualTo("value1");
     assertThat(fromJson.meta().value2).isEqualTo(2);
   }
+
+  @Test public void unwrapSecond() throws IOException {
+    final ServerSideLocationRequestMessage response = moshi.adapter(ServerSideLocationRequestMessage.class).fromJson("{\"foo\":{\"bar\":2,\"circle_id\":\"5530fa4915c1cf21e3043009\",\"circle\":\"5530fa4915c1cf21e3043009\"}}");
+    assertThat(response.bar()).isEqualTo(2);
+    assertThat(response.circleId()).isEqualTo("5530fa4915c1cf21e3043009");
+    assertThat(response.circle()).isEqualTo("5530fa4915c1cf21e3043009");
+  }
+
+  @AutoValue public abstract static class ServerSideLocationRequestMessage {
```

Comment:
To get the expected result it must be `@UnwrapJson("foo") ServerSideLocationRequestMessage` otherwise the adapter tries to unwrap each field as if the json would be:

{
  "foo" : { "bar": ...}
  "foo": { "circle": ...}
  //And so on
}